src/null-safety/migration-guide.md

```diff
@@ -119,14 +120,25 @@ then you can start migrating.
 Otherwise, use the **Resolvable** column to find
 null-safe releases, if they exist.
 
-For example, in the following output for a simple package,
-the green checkmarked versions support null safety:
+{{ site.alert.info }}
+  **Why do all dependencies need to support null safety?**
+  When all of an app's direct dependencies support null safety,
+  you can _run the app_ with sound null safety.
+  When all the dev dependencies support null safety,
+  you can _run tests_ with sound null safety.
+  You might also need null-safe dev dependencies for other reasons,
+  such as building.
```

Comment:
Not sure what building is here? Perhaps "code generation" is another likely example?